src/Build/Evaluation/LazyItemEvaluator.LazyItemOperation.cs

```diff
@@ -44,6 +45,11 @@ protected LazyItemOperation(OperationBuilder builder, LazyItemEvaluator<P, I, M,
                 _itemSpec.Expander = _expander;
             }
 
+            /// <summary>
+            /// Cache used for caching IO operation results
+            /// </summary>
+            protected ConcurrentDictionary<string, IEnumerable<string>> EntriesCache => _lazyEvaluator._entriesCache;
```

Comment:
@cdmihai Help me understand the lifetime of this cache? Just for a single Include?

Reply:
it spans the entire item evaluation phase, so covers all items outside of targets

---
In reply to: [143077078](https://github.com/Microsoft/msbuild/pull/2572#discussion_r143077078) [](ancestors = 143077078)